packages/zowe-explorer/src/Profiles.ts

```diff
@@ -1103,6 +1121,154 @@ export class Profiles extends ProfilesCache {
+                    name: baseProfile.name,
+                    merge: true,
+                    profile: updBaseProfile,
+                };
+
+                try {
+                    await profileManager.update(updateParms);
+                } catch (error) {
+                    vscode.window.showErrorMessage(
+                        localize("ssoLogin.unableToLogin", "Unable to log in. ") + error.message
+                    );
+                    return;
+                }
+                vscode.window.showInformationMessage(
+                    localize("ssoLogin.successful", "Login to API Mediation Layer was successful.")
+                );
+                this.allProfiles = this.allProfiles.map((item) => {
+                    item.profile.tokenValue = loginToken;
+                    return item;
+                });
+            } catch (error) {
+                vscode.window.showErrorMessage(
+                    localize("ssoLogin.unableToLogin", "Unable to log in. ") + error.message
+                );
+                return;
+            }
+        }
+    }
+
+    public async ssoLogout(node: IZoweNodeType) {
+        const baseProfile = await this.getBaseProfile();
+        const serviceProfile = node.getProfile();
+
+        // This check will handle service profiles that have username and password
+        if (serviceProfile.profile.user && serviceProfile.profile.password) {
+            vscode.window.showInformationMessage(
+                localize("ssoLogout.noBase", "Log out skipped. This profile does not support base profiles.")
+            );
+            return;
+        }
+
+        // This check is for optional credentials
+        if (
+            baseProfile &&
+            serviceProfile.profile.host &&
+            serviceProfile.profile.port &&
+            ((baseProfile.profile.host !== serviceProfile.profile.host &&
+                baseProfile.profile.port !== serviceProfile.profile.port) ||
+                (baseProfile.profile.host === serviceProfile.profile.host &&
+                    baseProfile.profile.port !== serviceProfile.profile.port))
+        ) {
+            vscode.window.showInformationMessage(
+                localize("ssoLogout.noBase", "Log out skipped. This profile does not support base profiles.")
+            );
+            return;
+        }
+
+        try {
+            const combinedProfile = await Profiles.getInstance().getCombinedProfile(serviceProfile, baseProfile);
+            const updSession = new Session({
+                hostname: combinedProfile.profile.host,
+                port: combinedProfile.profile.port,
+                rejectUnauthorized: combinedProfile.profile.rejectUnauthorized,
+                tokenType: combinedProfile.profile.tokenType,
+                type: combinedProfile.profile.type,
+                tokenValue: combinedProfile.profile.tokenValue,
+            });
+            await zowe.Logout.apimlLogout(updSession);
```

Comment:
can't have specific zowe calls outside of the api file, the api file in the common section could have new login and logout methods that could be utilized